kong/db/strategies/connector.lua

```diff
@@ -5,8 +5,11 @@ local fmt = string.format
 local Connector = {
   defaults = {
     pagination = {
-      page_size     = 512,  -- work with lmdb
-      max_page_size = 512,  -- work with lmdb
+      page_size     = 1000,
+      -- lmdb will not support huge page size
+      max_page_size = kong and kong.configuration and
+                      kong.configuration.database == "off" and
+                      2048 or 50000,
```

Comment:
Can we confirm the `2048` page size actually works? (i.e. `:page` should return at most `2048` entities at once)

Reply:
Added some asserts

Reply:
Test in post-func